code/game/machinery/big_manipulator.dm

```diff
@@ -276,7 +577,92 @@
 	icon = 'icons/obj/machines/big_manipulator_parts/big_manipulator_hand.dmi'
 	icon_state = "hand"
 	layer = LOW_ITEM_LAYER
+	appearance_flags = KEEP_TOGETHER | LONG_GLIDE | TILE_BOUND | PIXEL_SCALE
 	anchored = TRUE
 	greyscale_config = /datum/greyscale_config/manipulator_hand
 	pixel_x = -32
 	pixel_y = -32
+	/// We get item from big manipulator and takes its icon to create overlay.
+	var/datum/weakref/item_in_my_claw
+	/// Var to icon that used as overlay on manipulator claw to show what item it grabs.
+	var/mutable_appearance/icon_overlay
+
+/obj/effect/big_manipulator_hand/update_overlays()
+	. = ..()
+	if(isnull(item_in_my_claw))
+		icon_overlay = null
+		return
+	var/atom/movable/item_data = item_in_my_claw.resolve()
+	icon_overlay = mutable_appearance(item_data.icon, item_data.icon_state, item_data.layer, src, appearance_flags = KEEP_TOGETHER)
+	icon_overlay.color = item_data.color
+	icon_overlay.appearance = item_data.appearance
+	icon_overlay.appearance_flags = appearance_flags
+	icon_overlay.plane = ABOVE_MOB_LAYER
+	icon_overlay.pixel_x = 32 + calculate_item_offset("x")
+	icon_overlay.pixel_y = 32 + calculate_item_offset("y")
+	. += icon_overlay
+
+/// Updates item that is in the claw.
+/obj/effect/big_manipulator_hand/proc/update_claw(clawed_item)
+	item_in_my_claw = clawed_item
+	update_appearance()
+
+/// Calculate x and y coordinates so that the item icon appears in the claw and not somewhere in the corner.
+/obj/effect/big_manipulator_hand/proc/calculate_item_offset(coordinate)
+	var/offset
+	var/is_x_or = coordinate == "x"
+	switch(dir)
+		if(NORTH)
+			offset = is_x_or ? 0 : 32
+		if(SOUTH)
+			offset = is_x_or ? 0 : -32
+		if(EAST)
+			offset = is_x_or ? 32 : 0
+		if(WEST)
+			offset = is_x_or ? -32 : 0
+	return offset
+
+/// Priorities that manipulator use to choose to work on item with type same with what_type.
+/datum/manipulator_priority
+	/// Name that user will see in ui.
+	var/name
+	/// What type carries this priority.
+	var/what_type
+	/// Place in the priority queue. The lower the number, the more important the priority.
+	/// Doesn’t really matter what number you enter, user can set priority for themselves,
+	/// BUT!!!
+	/// Don't write the same numbers in the same parent otherwise something may go wrong.
```

Comment:
```suggestion
	/**
	 * Place in the priority queue. The lower the number, the more important the priority.
	* Doesn’t really matter what number you enter, user can set priority for themselves,
	* BUT!!!
	* Don't write the same numbers in the same parent otherwise something may go wrong.
	*/
```
You should know multiline comments.